blog/blogDevServer.ts

```diff
@@ -0,0 +1,89 @@
+import http from "http";
+import fs from "fs";
+import { watch } from "chokidar";
+import { type WebSocket, WebSocketServer } from "ws";
+import { buildFrontMatter } from "./buildFrontMatter";
+import { FRONT_MATTER_CACHE_FILENAME, POSTS_SOURCE_DIR } from "./paths";
+import { extractImports } from "./extractImports";
+import { parseMdx } from "./parser";
+import path from "path";
+
+let localCache: string = "[]";
```

Comment:
It's a string because that's what needs to be passed to `res.end()` even for JSON data.

But this is another situation that may not be relevant if a resource route can replace the dev server